service/routes.py

```diff
@@ -182,4 +182,22 @@ def get_item(customer_id, item_id):
         logger.error(f"Customer {customer_id} and corresponding item {item_id} could not be found.")
         abort(status.HTTP_404_NOT_FOUND, f"Customer {customer_id} and corresponding item {item_id} could not be found.")
  
-
+######################################################################
+# DELETE A SHOPCART
+######################################################################
+@app.route("/shopcarts/<int:customer_id>",, methods=["DELETE"])
+def delete_shopcart(customer_id):
+    """
+    Delete a shopcart
+    This endpoint will delete an shopcart based on the customer_id specified in the path
+    """
+    app.logger.info("Request to delete shopcart with id: %s", customer_id)
+    shopcart = ShopCarts.find_by_customer_id(customer_id)
+    if shopcart:
+        shopcart.delete()
+        app.logger.info("Shopcart with ID [%s] delete complete.", customer_id)
+    return "", status.HTTP_204_NO_CONTENT
+    # Abort if the shopcart does not exist
+    else:
```

Comment:
This `else` condition isnt necessary since you return 204 regardless of whether the record exists

Reply:
I think the return sentence should under the if condition. If the customer_id exists we delete it and return 204, if it doesn't exist return 404, is this logic correct?

Reply:
no, the `return` should be outside the if condition. 
If the `customer_id` exists, then you delete the item and return `204_NO_CONTENT`. 
even if the `customer_id `doesn't exist, you still return a `204_NO_CONTENT` since the item is already deleted. 
This ensures that the DELETE operation is idempotent. 

so you dont have to return a 404.

Reply:
oh I see, I removed the else condition and test. Thank you very much.